dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/hooks/AttributeValidationHook.java

```diff
@@ -129,7 +129,7 @@ protected void validateAttributeUniqueness( ValidationErrorReporter reporter,
         for ( UniqueAttributeValue uniqueAttributeValue : uniqueAttributeValues )
         {
             boolean isTeaUniqueInOrgUnitScope = !trackedEntityAttribute.getOrgunitScope()
-                || Objects.equals( organisationUnit.getUid(), uniqueAttributeValue.getOrgUnitId() );
+                || Objects.equals( organisationUnit.getUid(), uniqueAttributeValue.getOrgUnit() );
```

Comment:
This is a comparison between a String and a MetadataIdentifier, am I wrong?

Reply:
good point! thank you 🧷 Am updating it